server/tso/global_allocator.go

```diff
@@ -209,17 +209,41 @@ func (gta *GlobalTSOAllocator) syncMaxTS(ctx context.Context, dcLocationMap map[
 		if len(errList) > 0 {
 			return errs.ErrSyncMaxTS.FastGenWithCause(errList)
 		}
-		var syncedDCs []string
+		var (
+			respCount         int
+			syncedDCs         []string
+			inCollectingPhase bool
+		)
 		for resp := range respCh {
+			respCount++
 			if resp == nil {
 				return errs.ErrSyncMaxTS.FastGenWithCause("got nil response")
 			}
-			syncedDCs = append(syncedDCs, resp.GetDcs()...)
-			// Compare and get the max one
-			if resp.GetMaxLocalTs() != nil && resp.GetMaxLocalTs().GetPhysical() != 0 {
+			// Once we get a non-nil and non-zero MaxLocalTs first, we will think it's in the first phase
+			// of the Global TSO synchronization. So that we can have more detailed processing logic
+			// for each phase. For example, if we think we're in the first phase of the Global TSO
+			// synchronization, the inCollectingPhase will be set to true, and during this phase,
+			// any response with nil or empty MaxLocalTs will be regarded as an invalid response.
+			// Then the whole synchronization will fail.
+			if respCount == 1 && resp.GetMaxLocalTs() != nil && resp.GetMaxLocalTs().GetPhysical() != 0 {
+				inCollectingPhase = true
+			}
+			if inCollectingPhase {
+				// Handle the response of the first phase: collect all the Local TSOs
+				if resp.GetMaxLocalTs() == nil || resp.GetMaxLocalTs().GetPhysical() == 0 {
```

Comment:
`inCollectingPhase` means this condition always is `false`?

Reply:
and why respCount>=2 means not in  collecting phase?

Reply:
```go
if respCount == 1 && resp.GetMaxLocalTs() != nil && resp.GetMaxLocalTs().GetPhysical() != 0 {
	inCollectingPhase = true
}
```

This section means we will judge whether we are in the collecting phase (or the 1st phase) according to the first response we receive.